com.unity.mobile.notifications/Runtime/Android/Plugins/com/unity/androidnotifications/UnityNotificationManager.java

```diff
@@ -111,30 +111,30 @@ public UnityNotificationManager(Context context, Activity activity) {
         mBackgroundThread.start();
     }
 
-    public static UnityNotificationManager getNotificationManagerImpl(Context context) {
-        return getNotificationManagerImpl(context, (Activity) context);
+    static synchronized UnityNotificationManager getNotificationManagerImpl(Context context) {
+        if (mUnityNotificationManager == null) {
+            mUnityNotificationManager = new UnityNotificationManager();
+            mUnityNotificationManager.mVisibleNotifications = new HashSet<>();
+            mUnityNotificationManager.mScheduledNotifications = new ConcurrentHashMap();
+        }
+
+        // always assign context, as it might change
+        mUnityNotificationManager.mContext = context.getApplicationContext();
```

Comment:
Looks a bit unsafe, wouldn't it be better for mUnityNotificationManager to call context.getApplicationContext() when returning context instead of caching it?

Reply:
We've been caching it all the time, so it's quite safe. Here I'm considering scenarios like app update.